.github/workflows/CD.yml

```diff
@@ -517,50 +497,64 @@ jobs:
       #but using the tools in workflows from the branch triggered with workflow_dispatch.
       - uses: actions/checkout@v2
 
-      - name: Restore cached packages
+      - name: Cache if success
+        id: release-latest-image
         uses: actions/cache@v2
         with:
-          key: "${{ env.PACKAGE_CACHE_KEY }}"
-          path: build/packages
-
-      - name: Load Image
-        run: docker load < build/packages/$IMAGE_NAME.tar
+          path: |
+            VERSION
+          key: release-latest-image-${{ github.run_id }}
 
-      - name: Configure AWS Credentials
-        uses: aws-actions/configure-aws-credentials@v1
+      - name: Restore cached packages
+        if: steps.release-latest-image.outputs.cache-hit != 'true'
+        uses: actions/cache@v2
         with:
-          aws-access-key-id: ${{ secrets.RELEASE_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.RELEASE_SECRET }}
-          aws-region: us-west-2
+          key: "${{ env.PACKAGE_CACHE_KEY }}"
+          path: $PACKAGING_ROOT
 
       - name: Compare version with Dockerhub latest
         id: version
+        if: steps.release-latest-image.outputs.cache-hit != 'true'
         run: |
-          TAG=`cat build/packages/VERSION`
+          TAG="${{ needs.release-version-image.outputs.tag }}"
```

Comment:
This won't work because this job doesn't currently have `release-version-image` in its needs field.

Reply:
One more thing to learn since I thought the previous validation tests inherit the ```release-version-image``` so this step would also inherits it too. Fixed it